src/plugins/intel_gpu/src/graph/layout_optimizer.cpp

```diff
@@ -1493,7 +1493,7 @@ impl_types layout_optimizer::get_preferred_impl_type(program_node& node, format
             } else {
                 const auto scores_layout = nms_node.input_scores().get_output_layout();
                 if (scores_layout.is_dynamic()) {
-                    preferred_impl = impl_types::ocl;
+                    preferred_impl = impl_types::cpu;
```

Comment:
How does it affect performance? If there are any problems with ocl implementaion of NMS, shouldn't them be fixed, instead of forcing cpu impl?

Reply:
For the target model, performance will increase significantly (better than using patched ocl impl). But ocl implementation will also be fixed after it is fully tested.